app/scripts/lib/rpc-method-middleware/handlers/wallet-createSession/handler.ts

```diff
@@ -0,0 +1,250 @@
+
+type AbstractPermissionController = PermissionController<
+  PermissionSpecificationConstraint,
+  CaveatSpecificationConstraint
+>;
+
+/**
+ * Handler for the `wallet_createSession` RPC method which is responsible
+ * for prompting for approval and granting a CAIP-25 permission.
+ *
+ * This implementation primarily deviates from the CAIP-25 handler
+ * specification by treating all scopes as optional regardless of
+ * if they were specified in `requiredScopes` or `optionalScopes`.
+ * Additionally, provided scopes, methods, notifications, and
+ * account values that are invalid/malformed are ignored rather than
+ * causing an error to be returned.
+ *
+ * @param req - The request object.
+ * @param res - The response object.
+ * @param _next - The next middleware function.
+ * @param end - The end function.
+ * @param hooks - The hooks object.
+ * @param hooks.listAccounts - The hook that returns an array of the wallet's evm accounts.
+ * @param hooks.findNetworkClientIdByChainId - The hook that returns the networkClientId for a chainId.
+ * @param hooks.requestPermissionApprovalForOrigin - The hook that prompts the user approval for requested permissions.
+ * @param hooks.sendMetrics - The hook that tracks an analytics event.
+ * @param hooks.metamaskState - The wallet state.
+ * @param hooks.metamaskState.metaMetricsId - The analytics id.
+ * @param hooks.metamaskState.permissionHistory - The permission history object keyed by origin.
+ * @param hooks.metamaskState.accounts - The accounts object keyed by address.
+ * @param hooks.grantPermissions - The hook that grants permission for the origin.
+ */
+async function walletCreateSessionHandler(
+  req: JsonRpcRequest<Caip25Authorization> & { origin: string },
+  res: JsonRpcSuccess<{
+    sessionScopes: NormalizedScopesObject;
+    sessionProperties?: Record<string, Json>;
+  }>,
+  _next: JsonRpcEngineNextCallback,
+  end: JsonRpcEngineEndCallback,
+  hooks: {
+    listAccounts: () => { address: string }[];
+    findNetworkClientIdByChainId: NetworkController['findNetworkClientIdByChainId'];
+    requestPermissionApprovalForOrigin: (
+      requestedPermissions: RequestedPermissions,
+    ) => Promise<{ permissions: RequestedPermissions }>;
+    sendMetrics: (
+      payload: MetaMetricsEventPayload,
+      options?: MetaMetricsEventOptions,
+    ) => void;
+    metamaskState: {
+      metaMetricsId: string;
+      permissionHistory: Record<string, unknown>;
+      accounts: Record<string, unknown>;
+    };
+    grantPermissions: (
+      ...args: Parameters<AbstractPermissionController['grantPermissions']>
+    ) => Record<string, ValidPermission<string, Caveat<string, Json>>>;
+  },
+) {
+  const { origin } = req;
+  if (!isPlainObject(req.params)) {
+    return end(invalidParams({ data: { request: req } }));
+  }
+  const { requiredScopes, optionalScopes, sessionProperties } = req.params;
+
+  if (sessionProperties && Object.keys(sessionProperties).length === 0) {
```

Comment:
Do we need to validate `sessionProperties` more beyond this?

Reply:
Currently we don't use `sessionProperties` and [they are pretty wide open](https://github.com/ChainAgnostic/CAIPs/blob/9af528ed7efb9d67e894cd80af54263ab627a1de/CAIPs/caip-25.md?plain=1#L143) in what they can be...